File: 01_insertion_sort/ex02_ordem_decrescente.py
"""
Exercicio 02
Tema: 01_insertion_sort
"""

def ordenar_crescente(arr, tamanho):
    for i in range(1, tamanho):
        temp = arr[i]
        j = i - 1

        while j >= 0 and arr[j] < temp:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = temp
    
arr = [3, 1, 7, 4]

ordenar_crescente(arr, len(arr))

print(arr)

File: 01_insertion_sort/ex02_ordem_decrescente.c
/*
Exercicio 02
Tema: 01_insertion_sort
*/

void ordenar_decrescente(int * arr, int tamanho){
    for(int i = 1; i < tamanho; i++){
        int temp = arr[i];
        int j = i - 1;

        while(j >= 0 && arr[j] < temp){
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }
}

#include <stdio.h>

int main() {

    int arr[] = {3, 1, 7, 4};
    int tamanho = sizeof(arr) / sizeof(arr[0]);

    ordenar_decrescente(arr, tamanho);

    for(int i = 0; i < tamanho; i++){
        printf("%d ", arr[i]);
    }

    return 0;
}
